### DISCUSSION TITLE
.NET Core 8 and IBM JTOpen

### DISCUSSION THREAD

**Discussion by RuiPedroRasteiro** [ikvmnet/ikvm#452] [Q&A] [answered]
Hi,

I've been using IKVM to use IBM's Toolbox for Java ( https://github.com/IBM/JTOpen ) to communicate with iSeries/AS400 APIs via .NET Framework without issues using an old version of IKVM (7.2)

I'm trying to upgrade my software to .NET 8 and using IKVM 8.7.1 and JTOpen and I'm having issues related to the initialization of classes:

![image](https://github.com/ikvmnet/ikvm/assets/152307757/124658fe-17b9-4e00-b396-46dcdec56ec9)

Anyone using IKVM + JTOpen + .NET 8 successfully?

Thank you,
Rui

**Comment by wasabii** [comment]
Can you show the full stack trace?

**Comment by RuiPedroRasteiro** [comment]
at java.io.Writer..ctor(Object lock)
   at java.io.OutputStreamWriter..ctor(OutputStream out)
   at java.io.PrintWriter..ctor(OutputStream out, Boolean autoFlush)
   at com.ibm.as400.access.Trace..cctor()

![image](https://github.com/ikvmnet/ikvm/assets/152307757/c2a10b13-3696-4509-873b-bd995aa68db5)

PS: It also refers to a NullPointer Exception

Thank you!

**Reply by wasabii** [reply]
Is that it? No message down there?

**Reply by wasabii** [reply, marked as answer]
Welp, try creating a new java.lang.Object() before using any of the other libraries. Might be another bootstrapping error.

**Reply by RuiPedroRasteiro** [reply]
ah! it worked!

![image](https://github.com/ikvmnet/ikvm/assets/152307757/02be18ba-cf78-485e-969d-9041e6bcf5dd)

Thank you for your amazing work!
